tutorials/detecting-changes-in-sentinel-1-imagery-pt-2/tutorial.ipynb

```diff
@@ -0,0 +1 @@
efinite_. In the case of the $2\\times 2$ diagonal matrices that we are concerned with the eigenvalues are just the two diagonal elements themselves, so determining the Loewner order is trivial. For full $2\\times 2$ dual pol or $3\\times 3$ quad pol SAR imagery, devising an efficient way to determine the Loewner is more difficult, see [Nielsen et al. (2019)](https://ieeexplore.ieee.org/document/8736751).\n"," \n","So let's include the Loewner order in our change map:"]},{"cell_type":"code","metadata":{"id":"FOL0V1DNsCqH","colab_type":"code","colab":{}},"source":["c_map = p_value.multiply(0).where(p_value.lt(0.01),1)\n","diff = im2.subtract(im1)\n","d_map = c_map.multiply(0)                    # Initialize the direction map to zero.\n","d_map = d_map.where(det(diff).gt(0), 2)      # All pos or neg def diffs are now labeled 2.\n","d_map = d_map.where(diff.select(0).gt(0), 3) # Re-label pos def (and label some indef) to 3.\n","d_map = d_map.where(det(diff).lt(0), 1)      # Label all indef to 1.\n","c_map = c_map.multiply(d_map)                # Re-label the c_map, 0*X = 0, 1*1 = 1, 1*2= 2, 1*3 = 3."],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"MVh9eU6V3bVF","colab_type":"text"},"source":["Now we display the changes, with positive definite red, negative definite green, and indefinite yellow:"]},{"cell_type":"code","metadata":{"id":"b8_ggucVvOuh","colab_type":"code","colab":{}},"source":["mp = folium.Map(location=location, zoom_start=12, height=800, width=1000)\n","mp.add_ee_layer(crop2018, {min: 0, max: 255}, 'crop2018')\n","mp.add_ee_layer(c_map.updateMask(c_map.gt(0)), {'min': 0,'max': 3, 'palette': ['black','yellow','green','red']}, 'c_map')\n","mp.add_child(folium.LayerControl())"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"ghLYSYeLSj0P","colab_type":"text"},"source":["The more or less compact green changes indicate a decrease in reflectivity in both VV and VH bands, and correspond to crop harvesting (especially canola).\n","\n"]},{"cell_type":"markdown","metadata":{"id":"RvQO9S0W8UEQ","colab_type":"text"},"source":["### Outlook\n","We have now covered the subject of bitemporal change detection with GEE Sentinel-1 imagery. The beauty of GEE is that it is trivially easy to gather arbitrarily long time series of S1 images from the archive, all with revisit times of 6 or 12 days depending on whether one or both satellites are collecting data. The next part of the Tutorial will generalize the techniques we have learned so far to treat multitemporal change detection.\n"," \n","### Oh, and one more thing ...\n"," \n","We didn't mention it above, but note the similarity between Eq. (2.10) and Eq. (2.15). To go from the monovariate LRT to the bivariate LRT, we simply replace the product of  intensities $s_1s_2$ by the product of determinants $|c_1||c_2|$, the sum $s_1+s_2$ by $|c_1+c_2|$ and the factor $2^{2}$ by $2^4=2^{2\\cdot2}$. This observation will come in handy in Part 3."]}]}
```

Comment:
<p>Remove <code>import time</code>  since it was imported previously.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/earthengine-community/pull/222/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

Reply:
<p>Done - verified.</p>